flink-runtime/src/main/java/org/apache/flink/streaming/api/graph/util/OperatorInfo.java

```diff
@@ -0,0 +1,117 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.graph.util;
+
+import org.apache.flink.api.common.operators.ResourceSpec;
+import org.apache.flink.streaming.api.graph.StreamConfig;
+import org.apache.flink.streaming.api.graph.StreamEdge;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** A public class to help maintain the information of an operator */
+public class OperatorInfo {
+
+    private String chainedName;
+
+    private ResourceSpec chainedMinResource;
+
+    private ResourceSpec chainedPreferredResources;
+
+    private StreamConfig vertexConfig;
+
+    /**
+     * The config for node chained with the current operator, the key is the ID of the stream node
+     */
+    private final Map<Integer, StreamConfig> chainedConfigs;
+    /**
+     * This is used to cache the chainable outputs, to set the chainable outputs config after all
+     * job vertices are created.
+     */
+    private final List<StreamEdge> chainableOutputs;
+    /**
+     * This is used to cache the non-chainable outputs, to set the non-chainable outputs config
+     * after all job vertices are created.
+     */
+    private final List<StreamEdge> nonChainableOutputs;
+
+    public OperatorInfo() {
+        this.chainedConfigs = new HashMap<>();
+        this.chainableOutputs = new ArrayList<>();
+        this.nonChainableOutputs = new ArrayList<>();
+    }
+
+    public String getChainedName() {
+        return chainedName;
+    }
+
+    public void setChainedName(String chainedName) {
+        this.chainedName = chainedName;
+    }
+
+    public ResourceSpec getChainedMinResource() {
+        return chainedMinResource;
+    }
+
+    public void setChainedMinResource(ResourceSpec chainedMinResource) {
+        this.chainedMinResource = chainedMinResource;
+    }
+
+    public ResourceSpec getChainedPreferredResources() {
+        return chainedPreferredResources;
+    }
+
+    public void setChainedPreferredResources(ResourceSpec chainedPreferredResources) {
+        this.chainedPreferredResources = chainedPreferredResources;
+    }
+
+    public List<StreamEdge> getChainableOutputs() {
+        return chainableOutputs;
+    }
+
+    public void setChainableOutputs(List<StreamEdge> chainableOutputs) {
+        this.chainableOutputs.addAll(chainableOutputs);
+    }
+
+    public List<StreamEdge> getNonChainableOutputs() {
+        return nonChainableOutputs;
+    }
+
+    public void setNonChainableOutputs(List<StreamEdge> NonChainableOutEdges) {
```

Comment:
Non-static fields/methods must be in lower camel case. https://flink.apache.org/how-to-contribute/code-style-and-quality-formatting/#naming